tests/net/socket/tcp/src/main.c

```diff
@@ -368,6 +370,90 @@ static void restore_packet_loss_ratio(void)
 		"Error setting packet drop rate");
 }
 
+void test_v4_interface_down(void)
+{
+	/* Test if the data stops transmitting after the send returned with a timeout. */
+	int c_sock;
+	int s_sock;
+	int new_sock;
+	struct sockaddr_in c_saddr;
+	struct sockaddr_in s_saddr;
+	struct sockaddr addr;
+	socklen_t addrlen = sizeof(addr);
+
+	struct timeval optval = {
+		.tv_sec = 0,
+		.tv_usec = 500000,
+	};
+
+	struct net_stats before;
+	struct net_stats after;
+	uint32_t start_time, time_diff;
+	ssize_t recved;
+	int rv;
+	uint8_t rx_buf[10];
+
+	prepare_sock_tcp_v4(CONFIG_NET_CONFIG_MY_IPV4_ADDR, ANY_PORT,
+			    &c_sock, &c_saddr);
+	prepare_sock_tcp_v4(CONFIG_NET_CONFIG_MY_IPV4_ADDR, SERVER_PORT,
+			    &s_sock, &s_saddr);
+
+	test_bind(s_sock, (struct sockaddr *)&s_saddr, sizeof(s_saddr));
+	test_listen(s_sock);
+
+	test_connect(c_sock, (struct sockaddr *)&s_saddr, sizeof(s_saddr));
+	test_send(c_sock, TEST_STR_SMALL, strlen(TEST_STR_SMALL), 0);
+
+	test_accept(s_sock, &new_sock, &addr, &addrlen);
+	zassert_equal(addrlen, sizeof(struct sockaddr_in), "wrong addrlen");
+
+	rv = setsockopt(new_sock, SOL_SOCKET, SO_RCVTIMEO, &optval,
+			sizeof(optval));
+	zassert_equal(rv, 0, "setsockopt failed (%d)", errno);
+
+	test_recv(new_sock, MSG_PEEK);
+	test_recv(new_sock, 0);
+
+	/* At this point break the interface */
+	zassert_equal(loopback_set_packet_drop_ratio(1.0f), 0,
+		"Error setting packet drop rate");
+
+	test_send(c_sock, TEST_STR_SMALL, strlen(TEST_STR_SMALL), 0);
+
+	/* At this point break the interface */
+	start_time = k_uptime_get_32();
+
+	/* Message should never arrive */
+	recved = recv(new_sock, rx_buf, sizeof(rx_buf), 0);
+	time_diff = k_uptime_get_32() - start_time;
+
+	zassert_equal(recved, -1, "Unexpected return code");
+	zassert_equal(errno, EAGAIN, "Unexpected errno value: %d", errno);
+	zassert_true(time_diff >= 500, "Expected timeout after 500ms but "
+			"was %dms", time_diff);
+
+	/* Reading from client should indicate the socket has been closed */
+	test_eof(c_sock);
+
+	k_sleep(TCP_TEARDOWN_TIMEOUT);
```

Comment:
We should be relaly careful with those delays, it just increases the test time (with no justification in this case).

Reply:
You're right, removed